src/Xamarin.Android.Tools.AndroidSdk/Sdks/AndroidSdkWindows.cs

```diff
@@ -163,28 +165,24 @@ private static IEnumerable<string> GetOpenJdkPaths ()
 			}
 		}
 
-		/// <summary>
-		/// Locate OpenJDK installations by well known path.
-		/// </summary>
-		/// <returns>List of valid OpenJDK paths in version descending order.</returns>
-		private static IEnumerable<string> GetKnownOpenJdkPaths ()
+		static IEnumerable<string> GetMicrosoftOpenJdkFilesystemPaths ()
 		{
-			string JdkFolderNamePattern = "microsoft_dist_openjdk_";
+			const string JdkFolderNamePrefix = "jdk-";
 
 			var paths = new List<Tuple<string, Version>> ();
 			var rootPaths = new List<string> {
-				Path.Combine (Environment.ExpandEnvironmentVariables ("%ProgramW6432%"), "Android", "jdk"),
+				Path.Combine (Environment.ExpandEnvironmentVariables ("%ProgramW6432%"), "Microsoft"),
 				Path.Combine (Environment.GetFolderPath (Environment.SpecialFolder.ProgramFilesX86), "Android", "jdk"),
 			};
 
 			foreach (var rootPath in rootPaths) {
-				if (Directory.Exists (rootPath))  {
-					foreach (var directoryName in Directory.EnumerateDirectories (rootPath, $"{JdkFolderNamePattern}*").ToList ()) {
-						var versionString = directoryName.Replace ($"{rootPath}\\{JdkFolderNamePattern}", string.Empty);
-						if (Version.TryParse (versionString, out Version? ver)) {
-							paths.Add (new Tuple<string, Version>(directoryName, ver));
-						}
-					}
+				if (!Directory.Exists (rootPath))
+					continue;
+				foreach (var directoryName in Directory.EnumerateDirectories (rootPath, $"{JdkFolderNamePrefix}*").ToList ()) {
```

Comment:
Is the `.ToList()` needed? Seems like it will force the `EnumerateDirectories()` method to complete immediately.

Reply:
Nope; removed!